src/PracticesStore/PracticesStore.ts

```diff
@@ -0,0 +1,200 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import { makeAutoObservable, reaction, runInAction } from "mobx";
+
+import { Hydratable } from "../core/models/types";
+import {
+  CombinedUserRecord,
+  getUser,
+  OpportunityType,
+  searchClients,
+  subscribeToEligibleCount,
+  subscribeToOfficers,
+  UserRecord,
+} from "../firestore";
+import type { RootStore } from "../RootStore";
+import { Client } from "./Client";
+import { observableSubscription, SubscriptionValue } from "./utils";
+
+type ConstructorOpts = { rootStore: RootStore };
+
+export class PracticesStore implements Hydratable {
+  rootStore: RootStore;
+
+  isLoading = true;
+
+  error?: Error;
+
+  user?: CombinedUserRecord;
+
+  districtFilter: string[] = [];
+
+  officerFilter: string[] = [];
+
+  searchFilter?: string;
+
+  private compliantReportingEligibleCount?: SubscriptionValue<number>;
+
+  private clients?: Client[];
+
+  private officers?: SubscriptionValue<UserRecord[]>;
+
+  constructor({ rootStore }: ConstructorOpts) {
+    this.rootStore = rootStore;
+    makeAutoObservable(this, { rootStore: false });
+
+    // trigger some updates when filters change
+    reaction(
+      () => [this.officerFilter],
+      () => {
+        this.updateCaseloadSources();
+      }
+    );
+    reaction(
+      () => [this.searchFilter, this.officerFilter],
+      () => {
+        this.updateSearchSources();
+      }
+    );
+  }
+
+  /**
+   * Performs initial data fetches to enable Practices functionality and manages loading state.
+   * Expects user authentication to already be complete.
+   */
+  async hydrate(): Promise<void> {
+    this.isLoading = true;
+    this.error = undefined;
+    try {
+      const { user: authenticatedUser } = this.rootStore;
+
+      if (authenticatedUser && authenticatedUser.email) {
+        const userRecord = await getUser(authenticatedUser.email);
+        if (userRecord) {
+          runInAction(() => {
+            this.user = userRecord;
+            this.setDefaultCaseload(userRecord);
+            this.updateStateData(userRecord.info.stateCode);
+          });
+        } else {
+          throw new Error(
+            `Unable to retrive user record for ${authenticatedUser.email}`
+          );
+        }
+      } else {
+        // We expect the user to already be authenticated
+        throw new Error("Missing email for current user.");
+      }
```

Comment:
You don't need to take this suggestion, but I when there are multiple checks like this that can throw errors, I like to invert the outer ones to throw the error in the then-clause of the if-statement so we don't separate the logic from the error and need to balance checks with errors. For example:

```suggestion
      if (!authenticatedUser?.email) {
        // We expect the user to already be authenticated
        throw new Error("Missing email for current user.");
      }
      
      const userRecord = await getUser(authenticatedUser.email);
      if (userRecord) {
        runInAction(() => {
          this.user = userRecord;
          this.setDefaultCaseload(userRecord);
          this.updateStateData(userRecord.info.stateCode);
        });
      } else {
        throw new Error(
          `Unable to retrieve user record for ${authenticatedUser.email}`
        );
      }
```

You could also invert the `userRecord` check as well, but since it's so small of a clause, it's not the same sort of issue.

Reply:
yeah good call this is much cleaner